androidsvg/src/main/java/com/caverock/androidsvg/SVG.java

```diff
@@ -1683,32 +1683,45 @@ public String getId() {
 
    // One of the element types that can cause graphics to be drawn onto the target canvas.
    // Specifically: 'circle', 'ellipse', 'image', 'line', 'path', 'polygon', 'polyline', 'rect', 'text' and 'use'.
-   static abstract class GraphicsElement extends SvgConditionalElement implements HasTransform
+   public static abstract class GraphicsElement extends SvgConditionalElement implements HasTransform
    {
       Matrix  transform;
+      RectF rectF;
 
       @Override
       public void setTransform(Matrix transform) { this.transform = transform; }
 
       public abstract boolean contains(Point point);
+      public abstract float[] getXYPointsArray();
+
+
+      public RectF initRectFWithOffset(float offset) {
+         rectF = new RectF();
+         android.graphics.Path path = new android.graphics.Path();
+         float[] xyArray = getXYPointsArray();
+
+         for (int i = 0; i < xyArray.length; i++) {
+            float xPoint = xyArray[i];
+            i++; //move to y position
+            float yPoint = xyArray[i];
 
-      public RectF initRectF(float[] xyArray) {
-          RectF rectF = new RectF();
-          android.graphics.Path path = new android.graphics.Path();
+            if(i == 1) {//case for the first (x,y) pair
+               path.moveTo(xPoint, yPoint);
+            } else {
+               path.lineTo(xPoint, yPoint);
+            }
+         }
 
-          for (int i = 0; i < xyArray.length; i++) {
-              int xPoint = (int) xyArray[i];
-              i++; //move to y position
-              int yPoint = (int) xyArray[i];
+         if (transform != null) {
+            path.transform(transform);
+         }
 
-              if(i == 1) {//case for the first (x,y) pair
-                  path.moveTo(xPoint, yPoint);
-              } else {
-                  path.lineTo(xPoint, yPoint);
-              }
-          }
-          path.computeBounds(rectF, false);
-          return rectF;
+         path.computeBounds(rectF, false);
+         if (offset != 0) {
+            rectF.offsetTo(rectF.left * offset, rectF.top * offset);

```

Comment:
This method only allows either increments for rectF's left and rectF's top, or decrements for rectF's left and rectF's top. In other words, the method is not generic enough to allow individual manipulation of the offset in the x and y directions. Should we name it appropriately to reflect that (the current name is too generic IMO)?